src/middleware.ts

```diff
@@ -144,12 +144,19 @@ export const persist = <S extends State>(
     deserialize = JSON.parse,
   } = options || {}
 
+  const setItem = async () => storage.setItem(name, await serialize(get()))
+
+  const savedSetState = api.setState
+
+  api.setState = (state: PartialState<S>, replace?: boolean) => {
```

Comment:
The types are redundant
```suggestion
  api.setState = (state, replace) => {
```